File: app/src/main/java/com/live/ui/juhe/AnchorListActivity.java
package com.live.ui.juhe;

import android.os.Bundle;
import android.support.v4.widget.SwipeRefreshLayout;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.Toolbar;

import com.live.R;
import com.live.adpter.AnchorAdapter;
import com.live.api.JuheService;
import com.live.base.BaseActivity;
import com.live.entity.Anchor;
import com.live.network.RetrofitManager;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import io.github.prototypez.savestate.core.annotation.AutoRestore;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class AnchorListActivity extends BaseActivity {

    @BindView(R.id.rv_activity_anchor)
    RecyclerView rv_activity_anchor;
    @BindView(R.id.srl_activity_anchor)
    SwipeRefreshLayout srl_activity_anchor;
    @BindView(R.id.toolbar_activity_anchor)
    Toolbar toolbar_activity_anchor;


    private List<Anchor.ZhuboBean> list = new ArrayList<>();
    private AnchorAdapter adapter;

    @AutoRestore("addr")
    String addr;

    @AutoRestore("title")
    String title;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_anchor);
        ButterKnife.bind(this);

        adapter = new AnchorAdapter(list);
        rv_activity_anchor.setAdapter(adapter);

        addr = getIntent().getStringExtra("addr");
        title = getIntent().getStringExtra("title");

        toolbar_activity_anchor.setTitle(title);
        setSupportActionBar(toolbar_activity_anchor);


        srl_activity_anchor.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
            @Override
            public void onRefresh() {
                requestData();
            }
        });

        srl_activity_anchor.post(new Runnable() {
            @Override
            public void run() {
                srl_activity_anchor.setRefreshing(true);
                requestData();
            }
        });
    }


    private void requestData() {
        RetrofitManager.getInstance().getService(JuheService.class).getAnchor(addr).enqueue(new Callback<Anchor>() {
            @Override
            public void onResponse(Call<Anchor> call, Response<Anchor> response) {
                srl_activity_anchor.setRefreshing(false);
                Anchor temp = response.body();
                if (temp != null) {
                    list = temp.zhubo;
                    adapter.setNewData(list);
                }
            }

            @Override
            public void onFailure(Call<Anchor> call, Throwable t) {
                srl_activity_anchor.setRefreshing(false);
            }
        });
    }


}

File: app/src/main/java/com/live/ui/juhe/AnchorListActivity.kt
package com.live.ui.juhe

import android.os.Bundle
import com.live.R
import com.live.adpter.AnchorAdapter
import com.live.api.JuheService
import com.live.base.BaseActivity
import com.live.entity.Anchor
import com.live.network.RetrofitManager
import kotlinx.android.synthetic.main.activity_anchor.*
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import java.util.*

class AnchorListActivity : BaseActivity() {


    private var list: List<Anchor.ZhuboBean> = ArrayList()
    private var adapter: AnchorAdapter? = null

    private var addr: String? = null

    private var title: String? = null


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_anchor)

        adapter = AnchorAdapter(list)
        rv_activity_anchor!!.adapter = adapter

        addr = savedInstanceState?.getString("addr") ?: intent.getStringExtra("addr")
        title = savedInstanceState?.getString("title") ?: intent.getStringExtra("title")

        toolbar_activity_anchor!!.title = title
        setSupportActionBar(toolbar_activity_anchor)

        srl_activity_anchor!!.setOnRefreshListener { requestData() }

        srl_activity_anchor!!.post {
            srl_activity_anchor!!.isRefreshing = true
            requestData()
        }
    }

    override fun onSaveInstanceState(outState: Bundle?) {
        outState?.putString("title", title)
        outState?.putString("addr", addr)
        super.onSaveInstanceState(outState)
    }

    override fun onRestart() {
        super.onRestart()
        //从home点进来 跳轉到直播页
        if (BDLiveActivity.startBackground == 1) {
            startActivity(intent)
        }
    }

    private fun requestData() {
        RetrofitManager.getInstance()?.getService(JuheService::class.java)?.getAnchor(addr)?.enqueue(object : Callback<Anchor> {
            override fun onResponse(call: Call<Anchor>, response: Response<Anchor>) {
                srl_activity_anchor!!.isRefreshing = false
                val temp = response.body()
                if (temp != null) {
                    list = temp.zhubo
                    adapter!!.setNewData(list)
                }
            }

            override fun onFailure(call: Call<Anchor>, t: Throwable) {
                srl_activity_anchor!!.isRefreshing = false
            }
        })
    }
}
